ed_sensor_integration/src/laser/updater.cpp

```diff
@@ -0,0 +1,662 @@
+#include <ed/laser/updater.h>
+
+#include <iostream>
+
+#include <geolib/Shape.h>
+
+#include <ed/world_model.h>
+#include <ed/entity.h>
+#include <ed/update_request.h>
+
+#include <opencv2/imgproc/imgproc.hpp>
+
+#include "ed/convex_hull_calc.h"
+
+#include <ed/io/json_writer.h>
+
+#include "ed_sensor_integration/association_matrix.h"
+
+#include <tue/profiling/timer.h>
+
+/**
+ * Calculate an error based on the quality of a fit.
+ *
+ * @param e entity being fitted
+ * @param lrf laser range finder
+ * @param rel_pose candidate pose of the entity relative to the laser range finder
+ * @param sensor_ranges distances measured by the lrf
+ * @param model_ranges predicted measurement distances using the model sans e
+ * @param[out] num_model_points number of points used for
+ * @return double error
+ */
+double getFittingError(const ed::Entity& e, const geo::LaserRangeFinder& lrf, const geo::Pose3D& rel_pose,
+                       const std::vector<float>& sensor_ranges, const std::vector<double>& model_ranges,
+                       int& num_model_points)
+{
+    std::vector<double> test_model_ranges = model_ranges;
```

Comment:
`test_model_ranges` and `model_ranges` are not changed in this function correct? Then a const reference is enough. No need to copy the vector. Very inefficient.

When you do need a copy, just have it be a normal argument instead of a reference.

Reply:
test_model_ranges is changed as its reference is passed to RenderResult res. There it is changed (another entity is rendered) That is why this copy is needed